### PR TITLE
OOTB PDLC Happy Path documentation

### PR DESCRIPTION
**Pull request by christinaausley**
## Description

Document the OOTB PDLC, as per the following acceptance criteria/requirements from https://github.com/camunda/product-hub/issues/2355.

**Note:** Getting Started Guide work may be moved into a new PR to avoid blocking the rest of the PR content from release.

Closes https://github.com/camunda/web-modeler/issues/10299, closes https://github.com/camunda/web-modeler/issues/10092)

**Note: Two new issues created as result of this PR:**
* https://github.com/camunda/camunda-docs/issues/4182
* https://github.com/camunda/camunda-docs/issues/4183

### [CI/CD page](https://docs.camunda.io/docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd/)
- [X] Introduce the concept of OOTB vs integrated PDLCs
    * Explain the value of an OOTB PDLC (see "release notes" section above)
    * Explain the use cases for OOTB PDLCs vs integrated PDLC
    * OOTB is appropriate for low-risk processes where you need to move fast without investing engineering effort
    * Integrated PDLC is appropriate for business-critical processes where you have strict governance or quality requirements
 - [X] There should be 2 branches in the diagram- one OOTB that stays within the modeler and one integrated that hands off to the customer's tool in their version control system
 _TW: Not needed now that we have the diagram for the OOTB pipeline?_
- [X] Explain that process applications are the best solution for projects with CI/CD. Visual diffing and Git Sync (https://github.com/camunda/product-hub/issues/2036) are a powerful combination

### [Process applications](https://docs.camunda.io/docs/next/components/modeler/web-modeler/process-applications/)
- [x] Work the changes into the Idea and Purpose section
- [x] Explain the entire journey for process application development: creation, Play, versioning+diffing+review, and promotion 
    * During creation, you choose which dev cluster to use. This is persisted, removing the complexity from the deployment screen
    * Token simulation is useful during the design phase to get the process flow right
    * Play is useful for validating the behavior of the process once implementation details have been added. It's a faster, more flexible alternative to the "Deploy" or "Run" modal that's designed for the development phase
    * Versioning helps users track changes across multiple assets. You can then review the diff between two versions of a BPMN file in a visual editor. Other files can be reviewed in a Git-like code editor. Once done reviewing, it's best to create a comment in the main diagram
    * An admin should set the stages within the Modeler
    * ~Users can see when the process was last deployed to a particular stage~ This will be tackled later with https://github.com/camunda/web-modeler/issues/10231
    * When a deployment occurs, there's a comment that tracks who deployed which version to which stage.
- [x] * Describe how processes are governed
    * This iteration focuses on establishing the process and logging activities so admins can audit process activities
    * Change Control: Users can announce a review is complete by making a comment
    * Deployment Control: Deployments can only go to pre-set approved clusters. Only users with the right privileges can deploy, and only org admins can deploy to `prod`-tagged clusters. Each deployment action is logged with the user and stage it was deployed to
 - [x] Describe the limitations
    * Play is not yet GA and may not be visible in their deployment. It also has its own set of limitations
    * Once created, a process app version may not be changed. Specifically, they cannot be deleted or rolled back
    * Visual diffing is not yet supported for DMN, form, connector, or other filetypes
    * Commenting is not yet supported in the milestone view. Any comments about feedback, approval, or changes required must go through the main file view
    * Governance is not yet enforceable, and comments can be deleted.
    * Defining the cluster stages, identifying clusters by tags, or promoting them are not yet supported in Self-Managed

### [Milestones](https://docs.camunda.io/docs/next/components/modeler/web-modeler/milestones/)
 - [x] Mention that process apps can simplify this flow

### [Getting started guides](https://docs.camunda.io/docs/next/guides/orchestrate-human-tasks/#step-4-run-your-process)
 - [ ] Play, milestone creation, and review are steps before running it on the cluster
 _OUTSTANDING: requires more work & investigation, possibly move to another PR?_

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [x] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Comment by christinaausley** [comment]
Unassigning myself -- let me know how I can help and review 👍

**Review by mesellings — COMMENTED** [review]
No review as reassigned

File: docs/components/modeler/web-modeler/deploy-process-application.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/deploy-process-application.md:11]: I believe we're changing them from Zeebe clusters to Camunda clusters- please confirm

File: docs/components/modeler/web-modeler/deploy-process-application.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/deploy-process-application.md:20]: A minor comment: this feels verbose and hyper-specific. I don't have a concrete requirement or request for change so feel free to ignore it for now

**Comment by mesellings** [comment]
@HanselIdes @vpellegrino I have completed the first draft of the docs for the OOTB PDLC as follows:

Please review the new Process application docs that incorporate the pipeline work. I have split this into more appropriate pages, to chunk the content and make it easier for users to browse and be directed to. 

For easier review, you can preview the docs at:

https://preview.docs.camunda.cloud/pr-4144/docs/next/components/modeler/web-modeler/process-applications/

Please suggest any other reviewers you feel might be helpful/appropriate?

**Note:** The Getting Started Guide updates will need more work and investigation, setup etc, and in the interests of time I wanted to get the main pipeline documentation out for review as I am away August 26-30. It may be that we take the Getting Started Guide work out into a new issue/PR?

File: docs/components/modeler/web-modeler/deploy-process-application.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/deploy-process-application.md:33]: Heads-up: In 8.6, I will start pushing users to use Play for this purpose rather than Run

File: docs/components/modeler/web-modeler/process-application-pipeline.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/process-application-pipeline.md:3]: In this context, I define the process development lifecycle (PDLC) as all activities that go from discovery to improvement, were some activities like monitoring or analysis are performed in a different app like Operate or Optimize
I define the `deployment pipeline` as the activities to get a release candidate through the different stages/environments and quality gates out into production

File: docs/components/modeler/web-modeler/process-application-pipeline.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/process-application-pipeline.md:4]: This doesn't match my mental image of how the deployment pipeline fits as a part of the development lifecycle. The deployment pipeline does not affect other activities like design, implementation, or maintenance

File: docs/components/modeler/web-modeler/process-application-pipeline.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/process-application-pipeline.md:29]: the deployment stages are on the deployment pipeline - the dev lifecycle is much bigger than this

File: docs/components/modeler/web-modeler/process-application-pipeline.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/process-application-pipeline.md:62]: Token simulation is most valuable during the design phase because it ignores implementation details. By the time it's implemented, the user should use Play

File: docs/components/modeler/web-modeler/process-application-versioning.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/process-application-versioning.md:23]: Although _technically the functionality is a bulk milestone_, it functions as a version. Versions make more sense in the context. We'll improve this in Q4

File: docs/components/modeler/web-modeler/process-application-pipeline.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/process-application-pipeline.md:10]:
```suggestion
You can use the Web Modeler to quickly develop and progress low-risk process application releases through the stages of a standard development lifecycle.
```

Comment (HanselIdes) [reply docs/components/modeler/web-modeler/process-application-pipeline.md:10]: Here's one way to clean up the language -please check through the PR as I didn't comment on all instances of development lifecycle / pipeline

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md
Comment (HanselIdes) [inline docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:19]: This needs some conceptual language cleanup, but it's a good summary. Should I start using life cycle rather than lifecycle in the rest of my communications?

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md
Comment (HanselIdes) [inline docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:22]: Git Sync is more relevant to the integrated use case. Is that part of a different PR?

**Review by HanselIdes — CHANGES_REQUESTED** [review]

File: docs/components/modeler/web-modeler/process-application-pipeline.md
Comment (mesellings) [reply docs/components/modeler/web-modeler/process-application-pipeline.md:62]: Moved to the Model stage 👍

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md
Comment (mesellings) [reply docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:19]: No, stick with lifecycle for now, I'll change the docs - we've just discussed this in DevEx and I think we should stick with lifecycle (one word) in these docs.

File: docs/components/modeler/web-modeler/process-application-pipeline.md
Comment (mesellings) [reply docs/components/modeler/web-modeler/process-application-pipeline.md:29]: Edited: You can deploy and promote an application process version through the stages of a standard deployment pipeline (development, testing, staging, and production).

File: docs/components/modeler/web-modeler/deploy-process-application.md
Comment (mesellings) [reply docs/components/modeler/web-modeler/deploy-process-application.md:33]: Should I remove the "Run" section in this page then? If so, do we need to replace it with a "Play" section, or just point people towards the Play feature docs I would guess?

File: docs/components/modeler/web-modeler/process-application-versioning.md
Comment (mesellings) [reply docs/components/modeler/web-modeler/process-application-versioning.md:23]: Thanks - I have to document it as is, so technically have to say bulk milestone and so on, but as soon as we can, we can start using 'version' 👍

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md
Comment (mesellings) [reply docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:22]: Not AFAIK but it sounds like it should be an issue aligned with the Git Sync work to update this integrated guide. I'm just checking with Cole if he is aware of an issue as he documented Git Sync.

File: docs/components/modeler/web-modeler/deploy-process-application.md
Comment (mesellings) [reply docs/components/modeler/web-modeler/deploy-process-application.md]: I've removed the specification as it was superfluous anyway, changed to read "to a cluster"

File: docs/components/modeler/web-modeler/deploy-process-application.md
Comment (HanselIdes) [reply docs/components/modeler/web-modeler/deploy-process-application.md:33]: You don't need to make changes for now, but we'll need to revisit this for the 8.6.0 release

Between now and 8.6, it makes sense to deploy your process app since Play doesn't connect to a cluster
Starting with 8.6, we should tell people to use Play to run their process in a development cluster and only use the Deploy/Run modals for interacting with other stages

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md
Comment (HanselIdes) [reply docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:22]: I raised the question because git sync felt out of place. I was left wondering: "powerful for what"? I proposed a use case that it helps with below
```suggestion
For low-risk processes, you can use the Web Modeler [process application development pipeline](/docs/components/modeler/web-modeler/process-application-pipeline.md) to quickly develop and progress process application releases through the stages of a standard development life cycle. [Milestone comparison](/docs/components/modeler/web-modeler/milestones.md#compare-milestones) (Visual and XML diffing) and [Git Sync](/docs/components/modeler/web-modeler/git-sync.md) provide a powerful combination for collaborating between team members using both Web and Desktop Modelers.
```

FYI Git sync has 3 use cases:
* provide a detailed history of changes and backup (as files deleted from Web Modeler cannot be recovered)- relevant to all use cases
* integrate desktop and web modelers - relevant to all use cases
* integrate with a CI/CD pipeline - mutually exclusive with the out-of-the-box deployment stages and promotion flow

Knowing this information, would you change anything else?

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md
Comment (HanselIdes) [inline docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:19]: I raised the question on line 22 because git sync felt out of place. I was left wondering: "powerful for what"? I proposed a use case that it helps with below
```suggestion
For low-risk processes, you can use the Web Modeler [process application development pipeline](/docs/components/modeler/web-modeler/process-application-pipeline.md) to quickly develop and progress process application releases through the stages of a standard development life cycle. [Milestone comparison](/docs/components/modeler/web-modeler/milestones.md#compare-milestones) (Visual and XML diffing) and [Git Sync](/docs/components/modeler/web-modeler/git-sync.md) provide a powerful combination for collaborating between team members using both Web and Desktop Modelers.
```

FYI Git sync has 3 use cases:
* provide a detailed history of changes and backup (as files deleted from Web Modeler cannot be recovered)- relevant to all use cases
* integrate desktop and web modelers - relevant to all use cases
* integrate with a CI/CD pipeline - mutually exclusive with the out-of-the-box deployment stages and promotion flow

Knowing this information, would you change anything else?

**Review by HanselIdes — APPROVED** [review]
The docs meet all my requirements. I added one suggestion to further refine them

File: docs/components/modeler/web-modeler/deploy-process-application.md
Comment (mesellings) [reply docs/components/modeler/web-modeler/deploy-process-application.md:33]: Thanks for the heads up: I've created https://github.com/camunda/camunda-docs/issues/4182 for this work so it doesn't get missed.

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md
Comment (mesellings) [reply docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md]: The use cases info could be useful on the Git Sync page itself, but not sure if any of it fits here in the process applications docs? Could you provide more context please?

**Review by HanselIdes — DISMISSED** [review]
LGTM

**Comment by mesellings** [comment]
@christinaausley If you have time to take a look over the docs that would be great, but it is quite a big PR so i understand if you don't have time atm? On a side note, as you are still the PR owner I can't assign you for review, and I don't want to close the PR and lose the thread of comments etc, are you okay to just add comments or send me comments in Slack? Unless you have any other ideas of the best way to do the review - I guess this might have occurred before here?

File: docs/components/modeler/web-modeler/create-a-process-application.md
Comment (christinaausley) [inline docs/components/modeler/web-modeler/create-a-process-application.md:4]: I think this character length for the description is a bit too small -- can we add a description that's closer to 150?

File: docs/components/modeler/web-modeler/create-a-process-application.md
Comment (christinaausley) [inline docs/components/modeler/web-modeler/create-a-process-application.md:11]: Would add a purpose here -- for example, "Create a new process application to *insert purpose*."

File: docs/components/modeler/web-modeler/create-a-process-application.md
Comment (christinaausley) [inline docs/components/modeler/web-modeler/create-a-process-application.md:53]: Select **the** Create new dropdown?

File: docs/components/modeler/web-modeler/deploy-process-application.md
Comment (christinaausley) [inline docs/components/modeler/web-modeler/deploy-process-application.md:35]: Nitpick -- "Run your process application to test, debug, and observe how it performs in a live environment."

File: docs/components/modeler/web-modeler/deploy-process-application.md
Comment (christinaausley) [inline docs/components/modeler/web-modeler/deploy-process-application.md:60]: Maybe link to the documentation on BPMN, DMN, and forms here?

File: docs/components/modeler/web-modeler/process-application-pipeline.md
Comment (christinaausley) [inline docs/components/modeler/web-modeler/process-application-pipeline.md:25]: "Fix errors shown in Modeler"

File: docs/components/modeler/web-modeler/create-a-process-application.md
Comment (mesellings) [reply docs/components/modeler/web-modeler/create-a-process-application.md:4]: Thanks - done 👍

File: docs/components/modeler/web-modeler/create-a-process-application.md
Comment (mesellings) [reply docs/components/modeler/web-modeler/create-a-process-application.md:11]: Thanks - added 👍

File: docs/components/modeler/web-modeler/create-a-process-application.md
Comment (mesellings) [reply docs/components/modeler/web-modeler/create-a-process-application.md:53]: Good spot!

File: docs/components/modeler/web-modeler/create-a-process-application.md
Comment (mesellings) [reply docs/components/modeler/web-modeler/create-a-process-application.md:55]: Thanks - edited 👍

File: docs/components/modeler/web-modeler/deploy-process-application.md
Comment (mesellings) [reply docs/components/modeler/web-modeler/deploy-process-application.md:35]: Agreed, much better - edited 👍

File: docs/components/modeler/web-modeler/process-application-pipeline.md
Comment (mesellings) [reply docs/components/modeler/web-modeler/process-application-pipeline.md:24]: Greta spot - edited 👍

File: docs/components/modeler/web-modeler/process-application-pipeline.md
Comment (mesellings) [reply docs/components/modeler/web-modeler/process-application-pipeline.md:25]: Thanks - edited 👍

**Review by christinaausley — COMMENTED** [review]
@mesellings Feel free to approve and merge this when you're ready 👍

**Comment by mesellings** [comment]
One last thing from this ticket so I don't forget - I think we should look at each of these landing pages, and in particular the titles should match the linking better - e.g. when I click on reference, I get to a page called "Overview" for example?

**Review by mesellings — APPROVED** [review]
Self-approved after approval from Christina and Eric

**Comment by yT0n1** [comment]
~~christinaausley @HanselIdes @wollefitz @mesellings Any idea where I can find the original asset for  https://github.com/camunda/camunda-docs/pull/4144/changes#diff-36920c0c12dd0dcbc4b32d86aed4176618601dddc377292cc303015a26a6d6d2
I need to remove `main process` from it.~~  

never mind, i hacked it together myself :D

**Pull request merged**
